api.py

```diff
@@ -405,7 +402,7 @@ def clear_dataset(**kwargs):
         datasets = list_datasets(**kwargs)
         to_clear = []
         for dataset in datasets:
-            if dataset['name'] == kwargs['name'] or (kwargs['regex'] and re.match(kwargs['regex'], dataset['name'])):
+            if dataset.get('name', 0) == kwargs['name'] or (kwargs['regex'] and re.match(kwargs['regex'], dataset.get('name', 0))):
```

Comment:
I believe this is aiming for robustness, but it will fail on regex:

```
> re.match(".", 0)
TypeError: expected string or buffer
```
What about checking dataset up front?

`if dataset.get('name') and (dataset["name"] == kwargs['name'] or (kwargs['regex'] and re.match(kwargs['regex'], dataset['name']))):`

Reply:
I'll make a PR for this